src/snowflake/connector/nanoarrow_cpp/ArrowIterator/FixedSizeListConverter.cpp

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+//
+
+#include "FixedSizeListConverter.hpp"
+
+namespace sf
+{
+Logger* FixedSizeListConverter::logger =
+    new Logger("snowflake.connector.FixedSizeListConverter");
+
+FixedSizeListConverter::FixedSizeListConverter(ArrowArrayView* array)
+: m_array(array)
+{
+}
+
+void FixedSizeListConverter::generateError(const std::string& msg) const
+{
+  logger->error(__FILE__, __func__, __LINE__, msg.c_str());
+  PyErr_SetString(PyExc_Exception, msg.c_str());
+}
+
+PyObject* FixedSizeListConverter::toPyObject(int64_t rowIndex) const
+{
+  if (ArrowArrayViewIsNull(m_array, rowIndex))
+  {
+    Py_RETURN_NONE;
+  }
+
+  if (m_array->n_children != 1)
+  {
+    std::string errorInfo = Logger::formatString(
+        "[Snowflake Exception] invalid arrow element schema for fixed size "
+        "list: got (%d) "
+        "children",
+        m_array->n_children);
+    this->generateError(errorInfo);
+    return nullptr;
+  }
+
+  // m_array->length represents the number of fixed size lists in the array
+  // m_array->children[0] has a buffer view that contains the actual data of
+  // each list, back-to-back m_array->children[0]->length represents the sum of
+  // the lengths of the fixed size lists in the array.
+
+  ArrowArrayView* elements = m_array->children[0];
+  const auto length = elements->length / m_array->length;
+  PyObject* list = PyList_New(length);
+
+  const int64_t startIndexWithoutOffset = rowIndex * length;
+  for (int64_t i = 0; i < length; ++i)
+  {
+    const auto bufferIndexWithoutOffset = startIndexWithoutOffset + i;
+    switch (elements->storage_type)
+    {
+      case NANOARROW_TYPE_INT8:
+      case NANOARROW_TYPE_INT16:
+      case NANOARROW_TYPE_INT32:
```

Comment:
Should we disallow all but 32 bit int/float to be consistent with our support in the runtime or is there any automatic narrowing that occurs causing us to emit 16/8 bit values for arrow?

Reply:
I kept all the types in for completeness so that we don't have to release a new connector if the supported precision types change (and since they didn't require extra logic). I can remove the non 32-bit variants if that's preferable, however.

Reply:
Fine to leave them. Maybe add a comment that these are not in use as of <current python connector version> but may be used in the future. Is there any way we can get unit test coverage for these paths until it's possible to test them e2e?

Reply:
There unfortunately don't seem to be unit tests of the arrow deserialization code, so I'm not sure if there's a simple way to test these right now.